nilearn/regions/signal_extraction.py

```diff
@@ -116,9 +116,11 @@ def img_to_signals_labels(imgs, labels_img, mask_img=None,
         labels_data[np.logical_not(mask_data)] = background_label
 
     data = _safe_get_data(imgs)
+    target_datatype = np.float64
```

Comment:
Here's something I don't understand.
The tests pass for:
```python
target_datatype = np.float
target_datatype = np.float32
target_datatype = np.float64
```
They fail for:
```python
target_datatype = np.float32 if data.dtype.kind != 'f' else np.float64
```
Someone help me understand why @GaelVaroquaux @jeromedockes

Reply:
Can you point us out to the test failure?

Reply:
By the way, I think that the actual line that I would like is:
```suggestion
    target_datatype = np.float32 if data.dtype.kind == np.float32 else np.float64
```
In other terms: let float32 pass through, and all the rest is cast to float64

Reply:
I have done various combinations of this and when I do, one of the tests fails.
You can try it locally.
Maybe there is something incorrect about the tests?